backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/SessionWithInitializedTablesFactory.java

```diff
@@ -44,17 +49,28 @@ public SessionWithInitializedTablesFactory(ClusterConfiguration clusterConfigura
     private Session createSession(Cluster cluster, String keyspace) {
         Session session = cluster.connect(keyspace);
         try {
-            new CassandraTypesCreator(module, session)
-                .initializeTypes();
-            new CassandraTableManager(module, session)
-                .ensureAllTables();
+            Stream<Boolean> operations = Stream.of(createType(session), createTables(session));
+            if (operations.anyMatch(updated -> updated)) {
```

Comment:
You do set the version to Latest if AT LEAST ONE table or type is created.

Here is a little scenario:

 - We design a table migration
 - With a new table
 - We restart james
 - Here, James create one table
 - And... update to the last data version here (because one table is created) WITHOUT running the migration

Your condition should be `All tables and types are created` :)

Reply:
Completely agree